ui/src/preview/preview-table-item.tsx

```diff
@@ -62,33 +68,135 @@ export const PreviewTableItem: React.FC<PreviewTableItemProps> = ({
 
     const actionMenuRef = React.useRef(null);
 
+    const publicationGroupAssetAmount = tableEntry.publicationGroup
+        ? publicationAssetChangeAmounts.groupAmounts[tableEntry.publicationGroup?.id]
+        : undefined;
+
+    const [displayedPublicationStage, moveTargetStage] = publish
+        ? [PublicationStage.STAGED, PublicationStage.UNSTAGED]
+        : [PublicationStage.UNSTAGED, PublicationStage.STAGED];
+
+    const stagePublicationAssetAmount = publish
+        ? publicationAssetChangeAmounts.staged
+        : publicationAssetChangeAmounts.unstaged;
+
+    const tableEntryAsPublishRequestIds = createPublishRequestIdsFromTableEntry(
+        tableEntry.id,
+        tableEntry.type,
+    );
+
+    const tableEntryAsRevertRequestSource: RevertRequestSource = {
+        id: tableEntry.id,
+        type: tableEntry.type,
+        name: tableEntry.name,
+    };
+
+    const changeItemPublicationStage = () => {
+        const newStage = publish ? PublicationStage.UNSTAGED : PublicationStage.STAGED;
+
+        previewOperations.setPublicationStage.forSpecificChanges(
+            tableEntryAsPublishRequestIds,
+            newStage,
+        );
```

Comment:
Tässä ensisilmäystä miten tulkitsin järkeväksi passailla parentin parentista peräisin olevia funkkarireferenssejä. Mielipiteitä? Oma tulkinta, että näiden pitäisi oikeasti majailla vaan reduksissa, jolloin niitä ei tarvitsisi passailla lainkaan.

Reply:
Jatkokehityksenä nämä voisi tosiaan sulloa reduxiin. Siellähän kuitenkin muutenkin asuu tieto siitä mitä muutoksia on stagetettuna, niin on se hieman hullunkurista, että sen datan manipulointifunkkarit välitetään jostain kauempaa. Tämä taulukkokaan ei kuitenkaan mikään geneerinen komponentti ole. Tekisin tämän kuitenkin tosiaan jatkokehityksenä, nyt voisi olla hyvä saada nämä nykyiset mainiin että päästään testailemaan toiminnallisuutta.

Muuten tämä `preview-table-item.tsx` näyttää kyllä kivalta nykyään. Nice and clean! 👏